File: frontend/src/app/services/category/category.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { Category } from './../../classes/category';

const url = 'api/category'

@Injectable()
export class CategoryService {
    
    constructor( private http: HttpClient ) {
    }
  
    getAll() : Observable<Array<Category>> {
        return this.http.get(url).map((data : any) =>  {return data});
    }
    get(id)  {
        return this.http.get(url + '/' + id);
    }
    
    insert(data) {
        return this.http.post(url, data);
    }
    
    update(data) {
        return this.http.put(url, data);
    }
    
    delete(id) {
        return this.http.delete(url + '/' + id);
    }
    
    

}

File: src/main/java/com/rolas/studies/service/category/CategoryService.java
package com.rolas.studies.service.category;

import java.util.List;

import com.rolas.studies.entities.Category;
import com.rolas.studies.views.CategoryVW;

public interface CategoryService {
	public List<CategoryVW> getAll();
	public Category insert(Category c);
	public boolean update(Category c);
	public boolean delete(Integer id);
	public Category get(Integer id);
}
